File: internal/domain/flight.go
package domain

import "time"

type Flight struct {
	ID           string `gorm:"type:uuid;primaryKey" json:"id"`
	FlightNumber string `gorm:"uniqueIndex;not null" json:"flightNumber"`

	AircraftID           string `gorm:"type:uuid;not null;index" json:"aircraftId"`
	OriginAirportID      string `gorm:"type:uuid;not null;index" json:"originAirportId"`
	DestinationAirportID string `gorm:"type:uuid;not null;index" json:"destinationAirportId"`

	DepartureTime time.Time `gorm:"not null" json:"departureTime"`
	ArrivalTime   time.Time `gorm:"not null" json:"arrivalTime"`

	Status FlightStatus `gorm:"type:text;not null" json:"status"`
}

type FlightRepository interface {
}

File: frontend/web/types/flight.ts
export type CabinClass =
  | "economy"
  | "premium_economy"
  | "business"
  | "first";

export type FlightStatus =
  | "SCHEDULED"
  | "BOARDING"
  | "DEPARTED"
  | "LANDED"
  | "CANCELLED"
  | "DELAYED";

export interface FlightSearchRequest {
  departureAirportCode: string;
  arrivalAirportCode: string;
  departureDate: string;
  passengers: number;
  cabinClass: CabinClass;
  orderBy?: string;
}

export interface FlightSearchResponse {
  flightId: string;
  flightNumber: string;

  departureAirportCode: string;
  arrivalAirportCode: string;

  status: FlightStatus;

  departureTime: string;
  arrivalTime: string;

  durationMinutes: number;

  availableSeats: number;
  lowestPrice: number;
}

export interface FlightResponse {
  id: string;

  flightNumber: string;

  aircraftId: string;

  originAirportId: string;
  destinationAirportId: string;

  departureTime: string;
  arrivalTime: string;

  status: string;
}

